joblib/numpy_pickle.py

```diff
@@ -389,29 +300,37 @@ def dump(value, filename, compress=0, cache_size=100, protocol=None):
     using compression can significantly slow down loading. In
     addition, compressed files take extra extra memory during
     dump and load.
+
     """
     if compress is True:
-        # By default, if compress is enabled, we want to be using 3 by
-        # default
+        # By default, if compress is enabled, we want to be using 3 by default
         compress = 3
+
     if not isinstance(filename, _basestring):
         # People keep inverting arguments, and the resulting error is
         # incomprehensible
         raise ValueError(
-              'Second argument should be a filename, %s (type %s) was given'
-              % (filename, type(filename))
-            )
+            'Second argument should be a filename, %s (type %s) was given'
+            % (filename, type(filename))
+        )
+    if cache_size > 0:
+        # Cache size is deprecated starting from version 0.10
+        warnings.warn("Cache size is set to a value > 0 but won't be used.",
+                      DeprecationWarning, stacklevel=2)
```

Comment:
It would be great to have a test for making sure we get a DeprecationWarning, grep for catch_warnings in scikit-learn or nilearn. We may have helper functions too modeled after assert_raise_regex I don't remember for sure.